d3a_interface/device_validator.py

```diff
@@ -265,3 +267,85 @@ def validate_storage_device(**kwargs):
         raise D3ADeviceException(
             {"mis_configuration": [f"fit_to_limit & energy_rate_change_per_update "
                                    f"can't be set together."]})
+
+
+def _validate_rate(energy_rate):
+    if not CepSettings.ENERGY_RATE_RANGE.min <= energy_rate <= \
+           CepSettings.ENERGY_RATE_RANGE.max:
+        raise D3ADeviceException(
+            {
+                "misconfiguration": [f"energy_rate should be in between "
+                                     f"{CepSettings.ENERGY_RATE_RANGE.min} & "
+                                     f"{CepSettings.ENERGY_RATE_RANGE.max}."]})
+
+
+def _validate_rate_profile(energy_rate_profile):
+    for date, value in energy_rate_profile.items():
+        print(f"date: {date}")
+        if date == "filename":
```

Comment:
There could also be another entry in the profile that is not the "filename" entry, the header.
So, i suggest to duplicate or move the check for the date format from the d3a.strategy.read_user_profile here.

Reply:
Sorry, I already discussed offline with @spyrostz . If you have a deeper look, I have skipped validating uploaded_profile because of its complexity.
This should have already removed, so I will remove this check now